cmd/kubeadm/app/util/kubeconfig/kubeconfig_test.go

```diff
@@ -178,3 +178,45 @@ func TestWriteKubeconfigToDisk(t *testing.T) {
 		}
 	}
 }
+
+func TestClientSetFromFile(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Couldn't create tmpdir")
+	}
+	defer os.RemoveAll(tmpdir)
+
+	var writeConfig = []struct {
+		name        string
+		cc          configClient
+		ccWithToken configClientWithToken
+		expected    error
```

Comment:
I'd rename `expected` to `expectedError` or something similar, to get `if (err != nil && rt.expected == nil) || ... ` lines more readable.